### PR TITLE
Add validation for projection dimensions in Camera2D + projections

### PR DESCRIPTION
**Pull request by pushfoo**
TL;DR:
1. Leave full doc rework until a later PR
2. Add validation which prevents delayed `ZeroDivisionError` which @MiCurry discovered
3. Add tests for validation

### Changes

- [x] Add validation to `Camera2D.from_raw_data`
  - [x] Add validation
  - [x] Add tests
- [x] Rename internal `*Data` attributes for clarity per Discord discussion
    * `_data` -> `_camera_data`
    * `_projection` -> `_projection_data`
- [x] Add `near` and `far` validation to `__init__`
- [x] Add validation to `.projection`

### Why?

1. Any of the following in an ortho projection cause a divide be zero:
  * `left == right`
  * `top == bottom`
  * `near == far`
4. It's hidden until you call `Camera2D.use` for the first time

Example traceback sanitized from [the Discord thread](https://discord.com/channels/458662222697070613/1228115132345745468/1228166379933274173):
```
  File "E:\Projects\SpaceGame\space.py", line 443, in on_draw
    self.cameras[player].use()
  File "C:\Users\User\AppData\Roaming\Python\Python312\site-packages\arcade\camera\camera_2d.py", line 697, in use
    self._ortho_projector.use()
  File "C:\Users\User\AppData\Roaming\Python\Python312\site-packages\arcade\camera\orthographic.py", line 134, in use
    _projection = self._generate_projection_matrix()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\AppData\Roaming\Python\Python312\site-packages\arcade\camera\orthographic.py", line 106, in _generate_projection_matrix
    return Mat4.orthogonal_projection(*_true_projection, self._projection.near, self._projection.far)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\python312\Lib\site-packages\pyglet\math.py", line 627, in orthogonal_projection
    sx = 2.0 / width
         ~~~~^~~~~~~
ZeroDivisionError: float division by zero
```
### Questions

- [x] ~~Should `ZeroProjectionDimension` subclass `ZeroDivisionError` or `ValueError`?~~ It's a `ValueError` since the math hasn't taken place yet.

### Follow-up work

- Add validation to future properties after rework

### CODE REVIEW

**Comment by pushfoo** [comment]
After discussion, consensus from @DragonMoffon and users who've tried 3.0 cameras seems to be:
1. The current changes are necessary almost entirely as-is
2. Defer further property validation until after #2065 and related follow-ups
3. The only further change needed in this PR is swapping the signatures of `Camera2D.__init__` and `Camera2D.from_raw_data` by:
   * Swapping the signatures and internals
   * Updating the tests
   * Fixing the examples

**Review by DragonMoffon — APPROVED** [review]
Validation looks good, and renaming is good. There are issues, but they are with the code rather than the PR, and I will fix them later.
